### PR TITLE
Install terminfo data under /usr/share/julia

### PR DESCRIPTION
**Pull request by nalimilan**
Just like all other libraries, we don't want internal Julia files to mess with system files.

Introduced by https://github.com/JuliaLang/julia/pull/55411.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -410,7 +410,7 @@ endif
 	$(INSTALL_F) $(JULIAHOME)/contrib/julia.appdata.xml $(DESTDIR)$(datarootdir)/metainfo/
 	# Install terminal info database
 ifneq ($(WITH_TERMINFO),0)
-	cp -R -L $(build_datarootdir)/terminfo $(DESTDIR)$(datarootdir)
+	cp -R -L $(build_datarootdir)/terminfo $(DESTDIR)$(datarootdir)/julia
```
Comment (ararslan) [inline Makefile:413]: I think this and the other places might need trailing slashes? Also looks like this might need to be `mkdir -p`'d.

Comment (nalimilan) [reply Makefile]: I don't think the slash is needed, but adding it doesn't hurt so I've pushed a commit (also fixing a mistake).

No need to call `mkdir -p` as the dir is already created above when installing other things (Base, stdlibs...).

**Review by ararslan — APPROVED** [review]
Looks good to me, thanks!

**Comment by nalimilan** [comment]
Thanks!

**Pull request merged**